crates/red_knot_python_semantic/src/types/infer.rs

```diff
@@ -435,6 +435,11 @@ impl<'db> TypeInferenceBuilder<'db> {
         matches!(self.region, InferenceRegion::Deferred(_)) || self.deferred_state.is_deferred()
     }
 
+    /// Are we currently inferring types in a stub file?
+    fn is_stub(&self) -> bool {
```

Comment:
nit: I'd call this method `in_stub` rather than `is_stub`, because (as your doc-comment points out!) it asks the question "Are we _in_ a stub file?"

I'd also make it a method on `InferContext` rather than the `TypeInferenceBuilder`, as then we can use it from `unpacker.rs` as well as from `infer.rs`.

Reply:
There can also be a wrapper `TypeInferenceBuilder::in_stub()` method that just does `self.context.in_stub()`

Reply:
Good point. Applied both.